hazelcast/src/main/java/com/hazelcast/client/impl/spi/impl/ClientClusterServiceImpl.java

```diff
@@ -150,8 +151,11 @@ public long getClusterTime() {
     @Override
     public Client getLocalClient() {
         final ClientConnectionManager cm = client.getConnectionManager();
-        final ClientConnection connection = cm.getActiveConnections().iterator().next();
-        InetSocketAddress inetSocketAddress = connection != null ? connection.getLocalSocketAddress() : null;
+        InetSocketAddress inetSocketAddress = null;
+        Iterator<ClientConnection> iterator = cm.getActiveConnections().iterator();
+        if (iterator.hasNext()) {
+            inetSocketAddress = iterator.next().getLocalSocketAddress();
+        }
         UUID clientUuid = cm.getClientUuid();
         return new ClientImpl(clientUuid, inetSocketAddress, client.getName(), labels);
```

Comment:
composing the `ClientImpl ` object with a null address, is this not a problem?

Reply:
Well, We don't have another chance. And this pr is not changing the behaviour. We were passing null for the address when connection is null anyway.